integration-tests/src/test/java/org/zowe/apiml/integration/authentication/providers/RefreshTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.integration.authentication.providers;
+
+import io.restassured.RestAssured;
+import io.restassured.http.Cookie;
+import org.junit.jupiter.api.*;
+import org.zowe.apiml.security.common.config.AuthConfigurationProperties;
+import org.zowe.apiml.security.common.login.LoginRequest;
+import org.zowe.apiml.util.TestWithStartedInstances;
+import org.zowe.apiml.util.categories.*;
+import org.zowe.apiml.util.config.ConfigReader;
+import org.zowe.apiml.util.config.SslContext;
+import org.zowe.apiml.util.http.HttpRequestUtils;
+
+import java.net.URI;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
+import static org.apache.http.HttpStatus.*;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.isEmptyString;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNot.not;
+import static org.zowe.apiml.util.SecurityUtils.*;
+
+@GeneralAuthenticationTest
+@SAFAuthTest
+@zOSMFAuthTest
+public class RefreshTest implements TestWithStartedInstances {
+
+    public static final AuthConfigurationProperties authConfigurationProperties = new AuthConfigurationProperties();
+
+    private final static String USERNAME = ConfigReader.environmentConfiguration().getCredentials().getUser();

```

Comment:
Did you consider using GatewayRequests class?

Reply:
Because it doesn't fit here in my oppinion. I understand the DSL as a tool for conveniently making correct requests, here partially correct requests are tested, which the DSL doesn't support. The DSL also doesn't support operating on headers which is what is needed.

Reply:
I will refactor in follow-up PR